x/announce/keeper/genesis.go

```diff
@@ -0,0 +1,81 @@
+package keeper
+
+import (
+	"bytes"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/strangelove-ventures/hyperlane-cosmos/x/announce/types"
+)
+
+// InitGenesis initializes the hyperlane announce module's state from a provided genesis
+// state.
+func (k Keeper) InitGenesis(ctx sdk.Context, gs types.GenesisState) error {
+	var storedAnnouncements *types.StoredAnnouncements
+	var err error
+
+	for _, announcement := range gs.Announcements {
+		storedAnnouncements, err = k.getAnnouncements(ctx, announcement.Validator)
```

Comment:
Can we change GenesisAnnouncement's announcement from StoredAnnouncement -> StoredAnnouncements? This should greatly improve efficiency. We don't need this k.getAnnouncements or the next dozen or so lines of logic. Just do the k.setAnnouncement().